### PR TITLE
add preinstallation checking whether node can even run kata

### PR DESCRIPTION
**Pull request by pmores**
Aims to fix a problem where kata daemon will happily install kata even on nodes that fundamentally cannot run kata.

### CODE REVIEW

File: images/daemon/pkg/daemon/kata_check_node.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        return true
+    }
+
+    // try loading the module (as kata-daemon we're root by definition
+    // so permissions shouldn't be a concern)
+    cmd := exec.Command("modprobe", moduleName)
+    if output, err := cmd.CombinedOutput(); err != nil {
+        fmt.Printf("'modprobe %s' failed: %s", moduleName, output)
+        return false
+    }
```
Comment (fidencio) [inline images/daemon/pkg/daemon/kata_check_node.go:82]: To me, this is quite an intrusive thing to be done as part of the check.
Consider we can load the module, how do we ensure the module will loaded next time we boot the node up?

Does the operator do something to ensure the module is up and will ever be while we have kata installed? If not, I'd just error out and say, "hey, sorry, no can do."

**Review by fidencio — COMMENTED** [review]
I have left one comment and, please, take my review with a grain of salt as I'm not familiar with the operator code base. :-)

**Comment by bpradipt** [comment]
@pmores it will be good to explore usage of node feature discovery for this requirement.
Details
https://kubernetes-sigs.github.io/node-feature-discovery/stable/get-started/index.html
https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/get-started/features.html#cpu
https://github.com/openshift/node-feature-discovery

File: images/daemon/pkg/daemon/kata_check_node.go (hunk 1 above)
Comment (pmores) [reply images/daemon/pkg/daemon/kata_check_node.go:82]: @fidencio This is what upstream kata-check does.  I'm pointing this out just as a rationale of this PR's current form, I don't have a strong opinion of this and I'm open to doing it either way.

**Comment by jensfr** [comment]
We don't have a daemon any more and we (Pavel, I and others) are exploring other options to do this kind of check without it. For example the node feature discovery operator. So I'm closing this one for now , thank you for your work Pavel!

**Pull request closed without merging**